ddtrace/contrib/aiohttp/patch.py

```diff
@@ -1,39 +1,292 @@
+import asyncio
+import functools
+import logging
+import sys
 import wrapt
 
-from ...pin import Pin
 from ddtrace.util import unwrap
 
+from ...propagation.http import HTTPPropagator
+from ...pin import Pin
+from ...ext import http as ext_http
+from ..httplib.patch import should_skip_request
+import aiohttp.client
+from aiohttp.client import URL
 
 try:
     # instrument external packages only if they're available
     import aiohttp_jinja2
     from .template import _trace_render_template
-
-    template_module = True
 except ImportError:
-    template_module = False
+    _trace_render_template = None
+
+
+log = logging.getLogger(__name__)
+PY_35 = sys.version_info >= (3, 5)
+
+
+# NOTE: this will create a trace for the outer request, and a span for each
+#       connect (redirect), and optionally a span for the read of the body
+
+
+def _get_url_obj(obj):
+    url_obj = getattr(obj, 'url_obj', None)  # 1.x
+    if url_obj is None:
+        url_obj = obj.url  # 2.x
+
+    return url_obj
+
+
+def _set_request_tags(span, url):
+    if (url.scheme == 'http' and url.port == 80) or \
+            (url.scheme == 'https' and url.port == 443):
+        port = ''
+    else:
+        port = url.port
+
+    url_str = '{scheme}://{host}{port}{path}'.format(
```

Comment:
Try `url_str = str(url.with_query(None).with_fragment(None))`
Would you keep userinfo?
If not -- `url_str = str(URL.build(scheme=url.scheme, host=url.host, port=url.port, path=url.path))` could work.
Anyway you've missed a colon between `host` and `port`.

Reply:
copy/paste from httplib, so going to keep for find/replace sake